packages/calcite-components/src/components/combobox/combobox.tsx

```diff
@@ -1412,12 +1500,35 @@ export class Combobox
     this.textInput.value?.focus();
   }
 
+  private updateSelectAllState(): boolean {
```

Comment:
I think this function should be reworked into two functions. One that updates the selected all state and one that gets the value of it.